app/controllers/bills_controller.rb

```diff
@@ -1,7 +1,8 @@
 class BillsController < ApplicationController
   before_action :get_address, only: :create
   before_action :check_cart, :new_bill, only: :new
-  before_action :load_detail_bill, only: %i(new create)
+  before_action :logged_in_user, :load_detail_bill, only: %i(new create)
+  before_action :logged_in_user, :load_bill, only: :update
```

Comment:
```suggestion
  before_action :logged_in_user, only: %i(new create update)
  before_action :load_detail_bill, only: %i(new create)
  before_action :load_bill, only: :update
```
Nên ri em nhé
cơ mà việc check `logged_in_user` nên được gọi trước nhá- tức là nó nên được đặt ở dưới  line 1

Reply:
dạ vag